### DISCUSSION TITLE
How to integrate a smart car(Jetauto from Phantom Technology)into Edge Foundry and manage the devices on the car？

### DISCUSSION THREAD

**Discussion by vegetable-rocket** [edgexfoundry/community#288] [Q&A] [answered]
I would like to integrate a smart car (Jetauto from Phantom Technology) into Edge Foundry and manage the devices on the car (such as cameras). Is this feasible? If so, are there any relevant tutorials available? If you can provide assistance, I would greatly appreciate it.
![96003d16923e5638313a19fe0f222ba](https://github.com/edgexfoundry/community/assets/100068831/6a1059c7-faa6-443f-ac98-988db89619bd)
![Uploading 67be2f36fc718bae934929fcde4af0f.jpg…]()

**Comment by cloudxxx8** [comment]
if your camera supports Onvif, you can check https://docs.edgexfoundry.org/3.1/microservices/device/services/device-onvif-camera/General/

**Reply by vegetable-rocket** [reply]
Thanks！I checked and found that the camera does not support the ONVIF protocol. Is there any other way? Is it feasible to connect the trolley and then call the camera information?

**Reply by cloudxxx8** [reply, marked as answer]
You need to implement your own device service for the proprietary camera protocol

**Reply by vegetable-rocket** [reply]
Thanks

**Comment by vegetable-rocket** [comment]
Is there a solution to connect a smart car equipped with the ROS system?

**Reply by jiekechoo** [reply]
What kind of connection is this camera? If it is a USB camera, there is a repository [device-usb-camera](https://github.com/edgexfoundry/device-usb-camera) you can try.

And you need to integrate EdgeX Foundry to the ROS system by yourself.

**Reply by vegetable-rocket** [reply]
The main purpose is to integrate multiple robots onto the platform and then use the platform to centrally manage them.
However, there is currently no relevant information on connecting robots to this platform.

**Reply by jiekechoo** [reply]
To determine how to solve your problem, I have a few questions:

- How do you connect to the robots you're working with? For example, physical connections and communication protocols.
- How do you establish connections to/from the robots? Who acts as the master or slave?
- What type of data do you need to transfer? Telemetry or images?

By the way, EdgeX is an edge-computing framework, not a platform for managing devices, maybe you could find a IoT platform.

**Reply by vegetable-rocket** [reply]
1.Connect by connecting the robot's own WiFi
2.Use the no machine platform to access the car system desktop for operation
3.The data mainly includes information on the position of the car, image information observed by the camera, and so on
Thanks for your answer！
